test/rematch2.jl

```diff
@@ -286,6 +284,22 @@ end
     @test r == 2
 end
 
+@testset "infer positional parameters from Rematch2.fieldnames(T) 4" begin
+    # struct T207d
+    #     x; z; y
+    #     T207d(x, y) = new(x, y, y)
```

Comment:
to make this test a bit more robust, could we make sure field `z` has a different value than `y` - maybe `y+1`? or hardc oded `0`?